### DISCUSSION TITLE
WithDefaultHttpClient does not work for some languages

### DISCUSSION THREAD

**Discussion by RezaBagheri8** [scalar/scalar#4220] [Q&A]
hello everyone
I hope you are well.

i am using asp.net core + .net 9 + Scalar.AspNetCore 1.2.56 + Microsoft.AspNetCore.OpenApi 9.0.0

I encountered a problem while using Scalar.Asp.net core package.
when I use
`
app.MapScalarApiReference(options =>
    {
        options.WithDefaultHttpClient(ScalarTarget.CSharp, ScalarClient.HttpClient);
    });
`
it works well

but for example when I use

`
app.MapScalarApiReference(options =>
    {
        options.WithDefaultHttpClient(ScalarTarget.JavaScript, ScalarClient.Fetch);
    });
`
then it does not work and show me the curl request by default

**Comment by hanspagel** [comment]
Oh, thanks for the report! I created an issue: https://github.com/scalar/scalar/issues/4223
